### PR TITLE
requested changes

### PR DESCRIPTION
**Pull request by orion-git**
Thanks for creating the plugin!

- implemented requested changes from https://github.com/runelite/plugin-hub/pull/4830#issuecomment-1761550229
- abstracted gateway address fetching and ping logic into new methods (getDefaultGatewayAddress and pingGateway)
- added comprehensive null and shutdown state checks, and refined exception handling

### CODE REVIEW

**Comment by jonathangarelick** [comment]
Hi, thanks for submitting a PR. Everything looks good. Have you tested the changes?

File: src/main/java/com/jonathang/TickFixerPlugin.java [hunk 1]
```diff
@@ -61,14 +61,25 @@ private InetAddress getDefaultGatewayAddress() throws IOException {
 
     private void pingGateway() {
         if (failureCount.get() >= MAX_FAILURES) {
-            log.error("Failed to ping default gateway 10 times. Terminating.");
-            exec.shutdown();
-            return;
+            try {
+                InetAddress newAddress = getDefaultGatewayAddress();
+                if (!newAddress.equals(lastKnownAddress)) {
+                    log.debug("Gateway address changed. Resetting failure count.");
```
[the remark this answers is not in the document]
Comment (jonathangarelick) [reply src/main/java/com/jonathang/TickFixerPlugin.java:67]: Ok, let's revert that commit then, and if we get user feedback saying they want it, we can add it back in.

**Comment by phife-dawg** [comment]
Thanks guys for working on this!

**Comment by jonathangarelick** [comment]
@orion-git this looks good, thanks. Next time it might just be easier to use `git revert 2ff571d`. Cheers

**Pull request merged**